test/test_nested_tensor_masking.py

```diff
@@ -1,350 +1,976 @@
+                                torch.tensor(3)
+                            ])
+                        ]))
+
+    def test_ntftm_single_tensor_all_true_mask(self):
+        tensor = torch.tensor([[1]])
+        mask = torch.tensor(True)
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, nt.nested_tensor([torch.tensor([1])]))
+
+        mask = torch.tensor([True])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, nt.nested_tensor([torch.tensor([1])]))
+
+    def test_ntftm_multi_tensor_scalar_true_mask(self):
+        tensor = torch.tensor([[1], [2], [3]])
+        mask = torch.tensor(True)
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, 
+                        nt.nested_tensor([
+                            torch.tensor([1]),
+                            torch.tensor([2]),
+                            torch.tensor([3])
+                        ]))
+
+        # Extra dim
+        tensor = torch.tensor([[[1]], [[2]], [[3]]])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        expected_res1 = nt.nested_tensor([
+                            torch.tensor([[1]]),
+                            torch.tensor([[2]]),
+                            torch.tensor([[3]])
+                        ])
+        TestCase.assertEqual(self, res_nt, expected_res1)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=2)
+        expected_res2 = nt.nested_tensor([
+                            nt.nested_tensor([
+                                torch.tensor([1])
+                            ]),
+                            nt.nested_tensor([
+                                torch.tensor([2])
+                            ]),
+                            nt.nested_tensor([
+                                torch.tensor([3])
+                            ])
+                        ])
+        TestCase.assertEqual(self, res_nt, expected_res2)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=3)
+        expected_res3 = nt.nested_tensor([
+                            nt.nested_tensor([
+                                nt.nested_tensor([
+                                    torch.tensor(1)
+                                ])
+                            ]),
+                            nt.nested_tensor([
+                                nt.nested_tensor([
+                                    torch.tensor(2)
+                                ])
+                            ]),
+                            nt.nested_tensor([
+                                nt.nested_tensor([
+                                    torch.tensor(3)
+                                ])
+                            ])
+                        ])
+        TestCase.assertEqual(self, res_nt, expected_res3)
```

Comment:
For completeness you could add another assertRaises case that makes sure this fails for nested_dim=4 (similar to above tests)

Reply:
added